src/Engine/GameInstance.java

```diff
@@ -18,7 +19,7 @@
 
   HashMap<Integer, XYCoord> playerCursors = null;
 
-  public GameInstance(GameMap map, Commander[] cos)
+  public GameInstance(MapMaster map, Commander[] cos)
```

Comment:
All these places where you are passing a `MapMaster` around... is there any reason that couldn't still be `GameMap`?

Reply:
GameInstance and GameEvents need to have MapMaster because MapMaster never lies.
My idea was that GameActions and UI use GameMap, but final validation is done with MapMaster so the program doesn't decide to actually do something it's not supposed to.
For instance, a MoveEvent couldn't fail/preempt if it had the same fogged view that produced the path it's using.